splitgraph/engine/base.py

```diff
@@ -202,7 +203,7 @@ def delete_table(self, schema: str, table: str) -> None:
             )
         else:
             self.run_sql(
-                SQL("DROP FOREIGN TABLE IF EXISTS {}.{}").format(
+                SQL("DROP FOREIGN TABLE IF EXISTS {}.{} CASCADE").format(
```

Comment:
This is a biiit scary, I'd be more comfortable with this if it (or the caller) deleted the overlay components explicitly so that this raises an error instead of inadvertently cascading to something important.